src/components/FileUpload.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import '@/css/FileUpload.css';
+
+interface FileUploadProps {
+  uploadUrl: string;
+}
+
+const FileUpload: React.FC<FileUploadProps> = ({ uploadUrl }) => {
+  const [showModal, setShowModal] = useState(false);
+  const [selectedFiles, setSelectedFiles] = useState<File[]>([]);
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files) {
+      const newFiles = Array.from(event.target.files);
+      setSelectedFiles((prevFiles) => [...prevFiles, ...newFiles]);
+    }
+  };
+
+  const handleUpload = async () => {
+    // Once POST API req is out -- Currently WIP so cannot make any progress here
+    if (selectedFiles.length > 0) {
+      const formData = new FormData();
+      selectedFiles.forEach((file) => {
+        formData.append('files', file);
+      });
+
+      try {
+        const response = await fetch(uploadUrl, {
+          method: 'POST',
+          body: formData,
+        });
+
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+
+        const data = await response.json();
+        console.log('Upload successful:', data);
+        
+        setSelectedFiles([]);
+        handleClose();
+      } catch (error) {
+        console.error('Upload failed:', error);
+      }
+    } else {
+      alert('Please select files to upload');
+    }
+  };
+
+  const toggleModal = () => {
+    setShowModal(!showModal);
+  };
+
+  const handleClose = () => {
+    setShowModal(false);
+    setSelectedFiles([]);
+  };
+
+  return (
+    <div>
+      <button onClick={toggleModal}>Upload Files</button>
+
+      {showModal && (
```

Comment:
we should use mantine's components whenever possible!

[modal docs](https://mantine.dev/core/modal/)